### PR TITLE
DEPR: tz_convert in the Timestamp constructor raises

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] xref #23621
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/scalar/timestamp/test_timestamp.py [hunk 1]
```diff
@@ -671,10 +671,10 @@ def test_constructor_invalid_frequency(self):
             Timestamp("2012-01-01", freq=[])
 
     @pytest.mark.parametrize("box", [datetime, Timestamp])
-    def test_depreciate_tz_and_tzinfo_in_datetime_input(self, box):
+    def test_raise_tz_and_tzinfo_in_datetime_input(self, box):
         # GH 23579
         kwargs = {"year": 2018, "month": 1, "day": 1, "tzinfo": utc}
-        with tm.assert_produces_warning(FutureWarning):
+        with pytest.raises(ValueError, match="Cannot pass a datetime or Timestamp"):
             Timestamp(box(**kwargs), tz="US/Pacific")
```
Comment (jbrockmendel) [inline pandas/tests/scalar/timestamp/test_timestamp.py]: can we also have a case with the tzinfo kwarg

**Comment by jbrockmendel** [comment]
one test request, otherwise LGTM

**Review by WillAyd — APPROVED** [review]
lgtm outside of @jbrockmendel request

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -457,6 +457,7 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 - Removed previously deprecated keyword "n" from :meth:`DatetimeIndex.shift`, :meth:`TimedeltaIndex.shift`, :meth:`PeriodIndex.shift`, use "periods" instead (:issue:`22458`)
 - Changed the default value for the `raw` argument in :func:`Series.rolling().apply() <pandas.core.window.Rolling.apply>`, :func:`DataFrame.rolling().apply() <pandas.core.window.Rolling.apply>`,
 - :func:`Series.expanding().apply() <pandas.core.window.Expanding.apply>`, and :func:`DataFrame.expanding().apply() <pandas.core.window.Expanding.apply>` to ``False`` (:issue:`20584`)
+- Timezone converting a tz-aware ``datetime.datetime`` or :class:`Timestamp` with :class:`Timestamp` and the ``tz`` argument now raises a ``ValueError`` (:issue:`23621`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: can you re-word this a bit, hard to grok.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/scalar/timestamp/test_timestamp.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/timestamp/test_timestamp.py]: i think we need separate `pytest.raises` contexts (and the tzinfo kwarg is supposed to be a tzinfo object)

**Comment by jbrockmendel** [comment]
LGTM pending green

**Comment by mroeschke** [comment]
All green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**